realm/realm-library/src/main/java/io/realm/internal/fields/FieldDescriptor.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal.fields;
+
+import java.util.Arrays;
+import java.util.List;
+
+import io.realm.RealmFieldType;
+import io.realm.internal.ColumnInfo;
+import io.realm.internal.Table;
+
+
+/**
+ * Class describing a single field possible several links away.
+ */
+public abstract class FieldDescriptor {
+    public interface SchemaProxy {
+        boolean hasCache();
+
+        ColumnInfo getColumnInfo(String tableName);
+
+        long getNativeTablePtr(String targetTable);
+    }
+
+    @Deprecated
+    public static FieldDescriptor createLegacyDescriptor(Table table, String fieldDescription, boolean allowLink, boolean allowList) {
+        return new LegacyFieldDescriptor(table, fieldDescription, allowLink, allowList);
+    }
+
+    /**
+     * TODO:
+     * I suspect that choosing the parsing strategy based on whether there is a ref to a ColumnIndices
+     * around or not, is bad architecture.  Almost certainly, there should be a schema that has
+     * ColumnIndices and one that does not and the strategies below should belong to the first
+     * and second, respectively.  --gbm
```

Comment:
No offence and I am not against leaving name in the comment, but it looks quite old style :P Like something people did before git was invented. To know who left the comment there, `git blame` should just work.